pkg/controller/pravegacluster/upgrade.go

```diff
@@ -322,10 +322,78 @@ func (r *ReconcilePravegaCluster) syncControllerVersion(p *pravegav1alpha1.Prave
 	return true, nil
 }
 
+func (r *ReconcilePravegaCluster) syncStoreVersion(p *pravegav1alpha1.PravegaCluster) (synced bool, err error) {
+	//this is to check if the target image is above 07 and the current image is below 07
+	if p.Status.TargetVersion != "" && !util.IsVersionBelow07(p.Status.TargetVersion) && util.IsVersionBelow07(p.Status.CurrentVersion) {
+		return r.syncSegmentStoreVersionFrom06To07(p)
+	}
+	return r.syncSegmentStoreVersion(p)
+}
+
+func (r *ReconcilePravegaCluster) syncSegmentStoreVersionFrom06To07(p *pravegav1alpha1.PravegaCluster) (synced bool, err error) {
+	newsts := pravega.MakeSegmentStoreStatefulSet(p)
+	controllerutil.SetControllerReference(p, newsts, r.scheme)
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: newsts.Name, Namespace: p.Namespace}, newsts)
+	//this check is to see if the newsts is present or not if it's not present it will be created here
+	if err != nil {
+		*newsts.Spec.Replicas = 0
+		err = r.client.Create(context.TODO(), newsts)
+		if err != nil {
+			return false, err
+		}
+	}
+
+	oldsts := &appsv1.StatefulSet{}
+	name := p.Name + "-pravega-segmentstore"
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: name, Namespace: p.Namespace}, oldsts)
+	//this check is to see if the old sts is present or not
+	if err != nil {
+		err = r.client.Get(context.TODO(), types.NamespacedName{Name: newsts.Name, Namespace: p.Namespace}, newsts)
+		//this checks is to verify if the old sts is not present new sts should be present thus this ends upgrade and true is returned
+		if err != nil {
+			return false, err
+		}
+		return true, nil
+	}
+
+	//this check is run till the value of old sts replicas is greater than 0 and will increase two replicas of the new sts and delete 2 replicas of the old sts
+	if *oldsts.Spec.Replicas > 0 {
```

Comment:
Can we not check for <2 instead of 0....
If # replicas in oldsts is <=2, then just increase the replicas in NewSTS by 2 and delete the old STS. Before deleting old STS invoke `r.syncStatefulSetPvc(oldsts)`

Reply:
I have changed this.